extensions/amp-image-slider/0.1/amp-image-slider.js

```diff
@@ -708,45 +716,44 @@ export class AmpImageSlider extends AMP.BaseElement {
     observeWithSharedInOb(this.element, (inViewport) =>
       this.viewportCallback_(inViewport)
     );
-    // Extensions such as amp-carousel still uses .setOwner()
-    // This would break the rendering of the images as carousel
-    // will call .scheduleLayout on the slider but not the contents
-    // while Resources would found amp-imgs' parent has owner and
-    // refuse to run the normal scheduling in discoverWork_.
-    // SIMPLER SOL: simply always call scheduleLayout no matter what
-    const owners = Services.ownersForDoc(this.element);
-    owners.scheduleLayout(
-      this.element,
-      dev().assertElement(this.leftAmpImage_)
-    );
-    owners.scheduleLayout(
-      this.element,
-      dev().assertElement(this.rightAmpImage_)
-    );
+
+    const appendHints = () => {
+      this.container_.appendChild(this.hintLeftBody_);
+      this.container_.appendChild(this.hintRightBody_);
+    };
 
     this.registerEvents_();
 
+    if (this.containsAMPImages_) {
+      // Extensions such as amp-carousel still uses .setOwner()
+      // This would break the rendering of the images as carousel
+      // will call .scheduleLayout on the slider but not the contents
+      // while Resources would found amp-imgs' parent has owner and
+      // refuse to run the normal scheduling in discoverWork_.
+      // SIMPLER SOL: simply always call scheduleLayout no matter what
+      const owners = Services.ownersForDoc(this.element);
+      owners.scheduleLayout(this.element, dev().assertElement(this.leftImage_));
+      owners.scheduleLayout(
+        this.element,
+        dev().assertElement(this.rightImage_)
+      );
+
+      return Promise.all([
+        dev()
```

Comment:
So far all the examples and test do not mix AMP Images and HTMLImageElements. Perhaps consider allowing this.